datasets/inception_subset.py

```diff
@@ -0,0 +1,174 @@
+    def __init__(self, used_plants, experiment, root_dir, used_modalities, plants: List[int]):
+        self.labels = labels
+        self.experiment = experiment
+        self.root_dir = root_dir
+        self.modalities = used_modalities
+        self.data = used_plants
+        self.plants = plants
+        self.num_plants = len(plants)
+
+        print('self.plants  = ', self.plants)
+
+    def __len__(self):   # needs to be corrected to the number of items per plant for each modality
+        return self.num_plants
+
+    def __getitem__(self, idx):
+        plant = self.plants[idx % self.num_plants]
+        cycle = idx // self.num_plants
+        data = self.data[self.data.num_plants * cycle + plant]
+        data['plant'] = idx % self.num_plants
+        return data
+
+    @staticmethod
+    def random_split(modalities: Modalities):
+        indices = np.arange(modalities.num_plants)
+        plant_labels = np.asarray(labels[modalities.experiment])
+        #  https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.train_test_split.html
+        train_indices, test_indices = train_test_split(indices, train_size=parameters.train_ratio,
+                                                       stratify=plant_labels)
+        return InceptionSubset(modalities, train_indices, parameters.experiment, parameters.experiment_path,
+                               parameters.used_modalities), InceptionSubset(modalities, test_indices,
+                                                                            parameters.experiment,
+                                                                            parameters.experiment_path,
+                                                                            parameters.used_modalities)
+
+    @classmethod  # A generator
+    def cross_validation(cls: Type['InceptionSubset'], used_plants: Modalities) -> \
+            Iterator[Tuple['InceptionSubset', 'InceptionSubset']]:
+        # https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.StratifiedKFold.html
+        n_plants = parameters.n_plants
+        n_classes = parameters.n_classes
+        n_splits = parameters.n_splits
+        X = tuple(i for i in range(n_plants))  # Plants by order
+        y = np.array(labels[parameters.experiment])
+        skf = StratifiedKFold(n_splits=n_splits, shuffle=True, random_state=1)
+        skf.get_n_splits(X, y)
+        for train, test in skf.split(X, y):
+            train_ds = cls(used_plants, parameters.experiment, parameters.experiment_path,
+                           parameters.used_modalities, train)
+            test_ds = cls(used_plants, parameters.experiment, parameters.experiment_path,
+                          parameters.used_modalities, test)
+            yield train_ds, test_ds
+
+    @staticmethod
+    def leave_one_out(modalities: Modalities, plant_idx: int):
+        rest_idx = list(range(modalities.num_plants))
+        del rest_idx[plant_idx]
```

Comment:
- [ ] use list comprehension, also consider renaming to something like `rest_indices`
```suggestion
        rest_indices = [idx for idx in range(modalities.num_plants) if idx != plant_idx]
```